### ISSUE TITLE
logging on python side of webapp

### ISSUE THREAD

**Issue by mtholder** [OpenTreeOfLife/opentree#761]
careful use of logging on the server side (rather than the current print statements) could make it easier to diagnose problems. Just logging exceptions would help.
https://github.com/OpenTreeOfLife/phylesystem-api/blob/master/modules/api_utils.py#L237
is the function we use for phylesystem-api logging (note that it has to be in modules in web2py)

**Comment by jar398** [comment]
Not sure of the tradeoff relative to just using the apache error log? which
requires zero setup, and is easy to find.

**Comment by mtholder** [comment]
for example, this weekend one of the comments was causing an error. This was fixed in https://github.com/OpenTreeOfLife/opentree/commit/ed543f225e5e20422b3bfa3478b834f65052b68b

but if we had logging, it would have been trivial to track down.

Unfortunately, web2py's error tickets system is pretty bad IMHO. It does not capture all of the info.  I was able to identify where the exception was being generated. But we still don't know which comment was causing the error. or maybe it was a time out caused by over-using the github issue API., or something else...
If we had logged that exception, we'd probably have figured it out in seconds from the stacktrace.

**Comment by jar398** [comment]
No, I agree completely, my question was just where to put the log messages,
in /var/log/apache2/error.log, or in some other location.

**Comment by mtholder** [comment]
Ah OK. I see. Would we run into file locking or buffer corruption issues if both apache and web2py were using the same file?

**Comment by jar398** [comment]
I *thought* that std stderr for the web2py process magically ended up in
the apache error log. We certainly see a log of output now that looks like
it's just randomly written to stderr or stdout. But now that you ask I
don't really know. It would be part of the WSGI protocol I guess.

**Comment by mtholder** [comment]
cool. 
It looks like the default behavior for the logging module is stderr stream (see https://docs.python.org/2/library/logging.handlers.html ). So if we want things in error.log we should be able to just use the logger functions to control the verbosity, but not worry about specifying a file path for the log.
